### ISSUE TITLE
Browser is jumping up on first select opening

### ISSUE THREAD

**Issue by SoWhat-lv** [fnagel/jquery-ui#96]
Hello!

Selectmenu is acting really wired: when I enter a webpage and click on select, browser scrolls to top, then I have to scroll back down to see a select options. When I do it again without refreshing page, everything works fine. 

here I made a screen recording: http://www.youtube.com/watch?v=WtdBRZvthcc

I use:
jQuery 1.5.2
jQuery UI: 1.8.11
Selectmenu from: fnagel-jquery-ui-1.8.4-236-g40b8920.zip
Peek from: revsystems-jQuery-Peek-0452ed2.zip
Chrome and FF

**Comment by hakanb** [comment]
I encountered the same issue. I believe it's because of the href="#" on the anchor tag that replaces the dropdown.  

I was able to fix it by running e.preventDefault() when a.ui-selectmenu is clicked:

```
$("a.ui-selectmenu").bind("click", function(e) {
    e.preventDefault();
});
```

Not sure what other affects this may have, but thought I'd share.

**Comment by SoWhat-lv** [comment]
thanks hakanb, it works!

**Comment by fnagel** [comment]
So this problem is solved or needs further fixes?

**Comment by SoWhat-lv** [comment]
fnagel, I think you should try to fix it in next release. Because I don't have any non-standart configuration, so probably lots of other users also has this problem. If you need more information, feel free to ask ;)

**Comment by fnagel** [comment]
I will take a look as soon as possible. Issue reopened.

**Comment by fnagel** [comment]
This problem problem (the native click event with the empty anchor) should be fixed by the return false when click or mousedown event occurs and Im currently not able to reproduce the issue. 

Could you please provide a demo zip or use this: https://github.com/fnagel/jquery-ui/issues/61
Please try to replace the two return false; with the events with e.preventDefault(); in the selectmenu source (about line 90) and report if this works for you?

**Comment by SoWhat-lv** [comment]
No, this isn't working. I will later create a demo zip, but right now you can take a look at website http://bodypointmassage.tk

**Comment by fnagel** [comment]
Please use the "comment" not "comment & close" button :-)

Your website works perfectly for me when using FF4 on WIn XP SP3. (using the selectmenu widgets within the "Book your session" button). Please provide a miminal setup example.

**Comment by SoWhat-lv** [comment]
sorry. Yes, It works perfectly because I am using the code that hakanb gave me: $("a.ui-selectmenu").bind("click", function(e) {e.preventDefault();});

**Comment by fnagel** [comment]
Ahh ok, I did not look at the sourcecode. 

I will wait for the demo file.

**Comment by SoWhat-lv** [comment]
I commented this code for you :) now you should see the problem

**Comment by SoWhat-lv** [comment]
here is the zip http://dl.dropbox.com/u/2691093/selectmenu.zip

**Comment by shotcowboystyle** [comment]
This issue seems to occur for me in IE7-8. IE8 causes the page 'jump', where IE7 just shifts the options popup.

**Comment by shotcowboystyle** [comment]
Also, hakanb's fix did NOT work in my case.

**Comment by TroyMN** [comment]
This was a bit of an adventure, hakanb's fix worked for static select options. However I was using nested selects where items were dynamically added/removed based on user choices. The fix didn't work in this case and eventually I learned that I had to unbind and then rebind the e.preventDefault() handler. 

In case it helps someone else the following is working for me in various browsers (where 'level6' is the id of the dynamically modified source select): 
```$('#level6').selectmenu({wrapperElement: "<div class='search_basic'> </div>",maxHeight: 500});
   $('#level6').parent("fieldset").show();
   $('#level6-button').unbind("click").bind("click", function(e) {e.preventDefault();});

``````
Note that ```$('a.ui-selectmenu').unbind("click").bind("click", function(e) {e.preventDefault();});``` did NOT work even though it seems like it should have. I had to specifically bind to the generated id of the link tag.
``````

**Comment by fnagel** [comment]
Sorry about the long waiting time, but I had an surgery and was not able to work on the selectmenu widget the last few weeks. I will take a look asap, please be patient!

**Comment by cvaida** [comment]
@FluffyElmo: Did you tried 
<code>
$("a.ui-selectmenu").<b>live</b>("click", function(e) {
    e.preventDefault();
});
</code>

http://api.jquery.com/live/

**Comment by fnagel** [comment]
@SoWhat-lv 
Sorry, but you demo is hardly usable. Please take a look here how to easily post a demo: https://github.com/fnagel/jquery-ui/issues/61
Ive tried to use only <pre> $('select').selectmenu({ style: "dropdown", width: 278, maxHeight: 200 });</pre> everything works fine. Please provide a demo and tell me what you would like to do. It seems you should take a look at he dependency.html demo which helps you to understand how to use callbacks. All possible methods and callbacks are listed in the Wiki!

@All
Im still not able to reproduce this problem. Please post a demo: https://github.com/fnagel/jquery-ui/issues/61

**Comment by TroyMN** [comment]
I'll try and find time to create a demo this weekend, it's entirely possible I was/is doing something wrong.

@cvaida I did try the .live() function but I'm not sure if I tried it as you suggest. I tried a lot of things around that time:)

**Comment by fnagel** [comment]
@FluffyElmo
Thats great to hear! Without a demo Im not able to help you guys out...

**Comment by pro85** [comment]
I have the same bug. I fix it by editing open method.
I remove 
<code>.find('li:not(.' + self.widgetBaseClass + '-group):eq(' + this._selectedIndex() + ') a')[0].focus() </code>
code.

**Comment by fnagel** [comment]
Mhhh, this focus is a troublemaker atm. But it is needed to ensure a correct positioning in popup style.

@pro85
Are you able to provide a demo, so I can reproduce the issue? The prepared fiddle is currently incorrect please see the first comment: #61

**Comment by pro85** [comment]
@fnagel
U can see the bug when u after load page minimize window and click on select menu. The body will be scroll to bottom.

**Comment by fnagel** [comment]
What do you mean by "minimize" window? Make it smaller or really mimimize it?

**Comment by pro85** [comment]
It's strange bug. But i modeling him by one metod. U need 3 or more times resize window. For example minimize/maximize/minimize and after this click on select. window will scroll.

**Comment by fnagel** [comment]
Probably related: https://github.com/fnagel/jquery-ui/issues/128

**Comment by AsemRadhwi** [comment]
@fnagel:

I'm experiencing the same issue. I tried @hakanab's solution, but no luck. I reproduced a demo on http://test-issue-96.heroku.com/

If the select was way down and it was clicked, the browser will annoyingly jump to the top of the page. In my case, the select is wrapped in a jquery ui accordion. It works fine without the accordion.

**Comment by fnagel** [comment]
I hopefully solved this problem. Please test the latest commit and give feedback.

I will close this one as multiple issues are affected. Please see this #147 summery and give feedback there!

**Comment by fnagel** [comment]
I feel cautiously optimistic Ive found a solution for this whole flickering/jumping/positioning issue. Finally. See and comment #147
